sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java

```diff
@@ -321,7 +337,26 @@ private HttpPipeline createPipeline() {
         final String clientName = properties.getOrDefault("name", "UnknownName");
         final String clientVersion = properties.getOrDefault("version", "UnknownVersion");
 
-        httpPolicies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+        // Find applicationId to use
+        String logApplicationId = null;
+        if (httpLogOptions != null) {
+            logApplicationId = httpLogOptions.getApplicationId();
+        }
+
+        String clientApplicationId = null;
+        if (clientOptions != null && clientOptions.getApplicationId() != null) {
+            clientApplicationId = clientOptions.getApplicationId();
+        }
+
+        if (logApplicationId != null && clientApplicationId != null
+            && !logApplicationId.equalsIgnoreCase(clientApplicationId)) {
+            throw logger.logExceptionAsError(new IllegalStateException(
+                "'httpLogOptions.getApplicationId() and clientOptions.getApplicationId()' cannot be different."));
+        }
```

Comment:
I don't think this is necessary. If we find the app id in ClientOptions, there's no need to check HttpLogOptions at all. App id in ClientOptions will always have higher priority.

Reply:
I think we would like to inform client if client has set two separate value in `clientOptions.getApplicationId()`  and ` httpLogOptions.getApplicationId()`. This is probably a configuration mistake by user.